lib/blockchain/chaindb.js

```diff
@@ -1952,6 +1991,28 @@ function fromU32(num) {
   return data;
 }
 
+class NeutrinoState {
```

Comment:
Going to discuss this on telegram instead of github, I think this pool / chain / filterindexer path is getting too messy

Reply:
Chose option 2 of saving filter headers in cache.